src/provisioning/api.ts

```diff
@@ -349,7 +350,8 @@ export class ProvisioningApi {
 
         // Now do the token exchange
         try {
-            const response = await axios.get<{sub: string}>(`${url}/_matrix/federation/v1/openid/userinfo`, {
+            const requestUrl = new URL("/_matrix/federation/v1/openid/userinfo", url);
```

Comment:
```
> new URL("/_matrix/federation/v1/openid/userinfo", new URL("https://matrix.org"));
URL {
  href: 'https://matrix.org/_matrix/federation/v1/openid/userinfo',
  origin: 'https://matrix.org',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'matrix.org',
  hostname: 'matrix.org',
  port: '',
  pathname: '/_matrix/federation/v1/openid/userinfo',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

Reply:
I hate URL's API, thanks for documenting it here :)